ambry-api/src/main/java/com.github.ambry/account/AccountBuilder.java

```diff
@@ -95,11 +91,36 @@ public AccountBuilder setName(String name) {
    * @param status The id to set.
    * @return This builder.
    */
-  public AccountBuilder setStatus(AccountStatus status) {
+  public AccountBuilder status(AccountStatus status) {
     this.status = status;
     return this;
   }
 
+  /**
+   * Sets the snapshot version of the {@link Account} to build.
+   * @param snapshotVersion The version to set.
+   * @return This builder.
+   */
+  public AccountBuilder snapshotVersion(int snapshotVersion) {
```

Comment:
is there any reason to allow this? Wouldn't it be better if we did something along the lines of
1) If we use the main constructor - we start with the default
2) If we use the copy constructor - we increment

Would that break use cases? AFAIK, this builder is used only when mutating accounts (in prod code). There is one tool that uses this but perhaps can be changed.

Reply:
Unfortunately, Nuage needs to use the main constructor in all cases, since they are converting a rest.li resource (defined by PDSC) into an Ambry Account object.